app/scripts/services/Citations.svc.js

```diff
@@ -1,14 +1,52 @@
 'use strict';
 
-angular.module('yourStlCourts').factory('Citations', function ($resource) {
+angular.module('yourStlCourts').factory('Citations', function ($resource,$q) {
   var CitationResource = $resource('citations/:id');
 
   function find(parameters) {
-    return CitationResource.get(parameters).$promise;
+    var deferred = $q.defer();
+    CitationResource.get(parameters,function(citations){
```

Comment:
Instead of creating a deferred, you can just return `CitationResource.get` and inside of the resolve (success) callback, just return your modified array of citations. You do not need the reject (error) callback at all in this case. It'll make it a little less cluttered